internal/storage/device_queue.go

```diff
@@ -276,7 +276,7 @@ func GetNextDeviceQueueItemForDevEUIMaxPayloadSizeAndFCnt(db sqlx.Ext, devEUI lo
 			return DeviceQueueItem{}, errors.Wrap(err, "get next device-queue item error")
 		}
 
-		if qi.FCnt < fCnt || len(qi.FRMPayload) > maxPayloadSize || (qi.TimeoutAfter != nil && qi.TimeoutAfter.Before(time.Now())) {
+		if qi.FCnt > fCnt || len(qi.FRMPayload) > maxPayloadSize || (qi.TimeoutAfter != nil && qi.TimeoutAfter.Before(time.Now())) {
```

Comment:
This is not correct. `fCnt` holds the next expected frame-counter and `qi.FCnt` holds the frame-counter of the queue item. The queue-item must always be >= `fCnt` thus when `qi.FCnt < fCnt`, it must be discarded.